src/scripts/domain/board.ts

```
@@ -133,14 +133,14 @@ export class Board implements BoardInfo {
 
-    if (newCell !== null) {
+    if (!isAttackMove && newCell !== null) {
       throw new Error(`Cell at ${to.x} ${to.y} not empty`);
     }
 
-    this._board[to.y][to.x] = piece;
     this._board[from.y][from.x] = null;
+    this._board[to.y][to.x] = piece;
 
-    piece.moveTo(to);
+    piece!.moveTo(to);
```

Biome (1.9.4)

[error] Forbidden non-null assertion.

Unsafe fix: Replace with optional chain operator ?. This operator includes runtime checks, so it is safer than the compile-only non-null assertion operator


(lint/style/noNonNullAssertion)